kafka/consumer_test.go

```diff
@@ -45,11 +44,17 @@ func TestConsumerAPIs(t *testing.T) {
 	if err != nil {
 		t.Fatalf("%s", err)
 	}
-
 	t.Logf("Consumer %s", c)
 
-	err = c.Subscribe("gotest", nil)
-	if err != nil {
+	testConsumerAPIs(t, c, nil)
+
+	// testing the API's with closed consumer.
+	testConsumerAPIs(t, c, getOperationNotAllowedErrorForClosedClient())
+}
+
+func testConsumerAPIs(t *testing.T, c *Consumer, errCheck error) {
+	err := c.Subscribe("gotest", nil)
+	if err != errCheck {
 		t.Errorf("Subscribe failed: %s", err)
```

Comment:
... Failed to return expected result %v : %v\n", errCheck, err

^ something like this here and in other instances

Reply:
It's already been done in case of producer tests, you can use that too